### PR TITLE
Prevent Panics in Field Trie Helpers

### PR DESCRIPTION
**Pull request by rauljordan**
**What type of PR is this?**

> Bug fix

**What does this PR do? Why is it needed?**

Part of #7558, this PR prevents panics in our beacon state field trie helpers by returning errors instead. However, it is still not clear how the input that leads to the panic originates in initial sync. At first sight, it seems like a clear sign of underflow/overflow we should watch out for. cc @farazdagi

### CODE REVIEW

File: beacon-chain/state/field_trie.go [hunk 1]
```diff
@@ -213,17 +219,22 @@ func handleEth1DataSlice(val []*ethpb.Eth1Data, indices []uint64, convertAll boo
...
-			return nil, err
+	if len(val) > 0 {
+		for _, idx := range indices {
+			if idx > uint64(len(val))-1 {
+				return nil, fmt.Errorf("index %d greater than number of items in eth1 data slice %d", idx, len(val))
+			}
+			err := rootCreator(val[idx])
+			if err != nil {
+				return nil, err
+			}
```
Comment (prestonvanloon) [inline beacon-chain/state/field_trie.go:237]: This looks like duplicated code from handleByteArrays. Can you abstract this logic to a common method?

File: beacon-chain/state/field_trie.go [hunk 2]
```diff
@@ -246,17 +257,22 @@ func handleValidatorSlice(val []*ethpb.Validator, indices []uint64, convertAll b
...
-	for _, idx := range indices {
-		err := rootCreater(val[idx])
-		if err != nil {
-			return nil, err
+	if len(val) > 0 {
+		for _, idx := range indices {
+			if idx > uint64(len(val))-1 {
+				return nil, fmt.Errorf("index %d greater than number of validators %d", idx, len(val))
+			}
+			err := rootCreator(val[idx])
```
Comment (prestonvanloon) [inline beacon-chain/state/field_trie.go:272]: Same here, these all look like the same code

**Comment by rauljordan** [comment]
@prestonvanloon these are different due to the input types :(. Wish we had generics, but abstracting it won't work at the moment

**Comment by prestonvanloon** [comment]
> @prestonvanloon these are different due to the input types :(. Wish we had generics, but abstracting it won't work at the moment

Sad :(

My only other feedback is that there are typos with "creater" it should be "creator".

**Review by nisdas — COMMENTED** [review]
I dont think this is the right place to have this check it should ideally be outside the field converters.  They are all the same array index validity checks. Also this kind of masks the original issue. We somehow got an invalid validator index from our state transition which looks like an overflow.

**Review by prestonvanloon — DISMISSED** [review]
There are still a typo or two in `rootCreater`, but approving anyway.

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**